hipcheck/src/analysis/result.rs

```diff
@@ -142,24 +131,21 @@ impl ThresholdPredicate {
 	}
 }
 
-fn pass_threshold<T: PartialOrd>(a: &T, b: &T, ord: &Ordering) -> Result<bool> {
-	a.partial_cmp(b)
-		.ok_or(hc_error!("threshold comparison failed for unknown reason"))
-		.map(|x| x == *ord)
+fn pass_threshold<T: Ord>(a: &T, b: &T, ord: &Ordering) -> bool {
```

Comment:
I just checked, and `Ordering` impls `Copy`, and because it's only a 3-element enum is likely smaller than a pointer anyway. I recommend changing this to take `Ordering` instead of `&Ordering`, which also means you could remove the dereference from the function body.